### ISSUE TITLE
Promote meck_history:result to meck API...?

### ISSUE THREAD

**Issue by rogerl-happening** [eproxus/meck#249]
I was looking for a way to grab the result of a mocked call. Currently, I'm just peeking into `meck:history()`.

So I went to implement something like `meck:capture(..., result)`, when I noticed that `meck_history:result()` exists. It was added in https://github.com/eproxus/meck/pull/163.

But it was demoted from the top-level `meck` API. I think this was a mistake -- it ought to be at the same level of abstraction as `meck:capture`.

Can we put it back? Or can we make `meck:capture` take `ArgNum | result`...?

**Comment by eproxus** [comment]
Yeah, good point. There is a risk of polluting the API with a lot of very specific functions that could perhaps be generalized a bit.

What do you think about adding a new function that somehow grabs a function call from the history using some nice API (like `capture/5,6` or similar?) and then returns a tuple with `{Args, Result}` or something?

**Comment by eproxus** [comment]
I guess I'm arguing for soft-deprecating `capture` in favor of something more generic.

I see it as `meck:history/1+2` being the firehose, and `capture/5+6` is a very specific subset of that ("give me _one_ call and _that_ particular argument value from it). It feels like there's something missing in-between, like "give me the arguments and return value from 'that particular call'", if that makes sense.

```erl
meck:history().
% [{<0.332.0>,{foo,bar,[a,b]},ok},
% {<0.332.0>,{foo,bar,[c,d]},ok},
% {<0.332.0>,
%  {foo,baz,[wat]},
%  error,some_error,
%  [{meck_ret_spec,eval_result,4,[{file,"src/meck_ret_spec.erl"},{line,93}]},
%   {foo,baz,[wat]}]},
% {<0.345.0>,{foo,bar,[x,y]},ok}]

meck:calls(foo, bar, 2).
% [{<0.332.0>,{foo,bar,[a,b]},ok},
% {<0.332.0>,{foo,bar,[c,d]},ok},
% {<0.345.0>,{foo,bar,[x,y]},ok}]

meck:call(foo, bar, 2, last).
% {<0.345.0>,{foo,bar,[x,y]},ok}

meck:call(foo, bar, 2, first).
% {<0.345.0>,{foo,bar,[a,b]},ok}

meck:calls(foo, bar, ['_', d])
% [{<0.332.0>,{foo,bar,[c,d]},ok}]
meck:call(foo, bar, ['_', d]) % implies 'last'
% {<0.345.0>,{foo,bar,[x,y]},ok}
```
(Totally made up proposal, so names and arguments are not that seriously suggested, yet).

With something like that, if you want the first argument and the result you could do:

```erl
{_From, [First|_], Result} = meck:call(foo, bar, ['_', d])
```

To get the nth argument from the argument list, you could use `lists:nth/2` for example.

**Comment by eproxus** [comment]
Another long-standing idea I have for a new version of the history API is to include timestamps too, but maybe this is scope creep at this time ☺️
